### PR TITLE
Support two argument at-irrational

### PR DESCRIPTION
**Pull request by LilithHafner**
```julia
@irrational logten 2.302585092994046 log(big(10))
# can now become
@irrational logten log(big(10))
```

Needs tests; I'll add them if folks think this is a good idea.

### CODE REVIEW

File: base/irrationals.jl [hunk 1]
```diff
@@ -165,13 +165,21 @@ end
...
+of `BigFloat`s given by the expression `def`. 
+    
+Optionally provide a pre-computed `Float64` value `val` which must equal `Float64(def)`.
+`val` will be computed automatically if omitted.
 """
 macro irrational(sym, val, def)
+    irrational(sym, val, def)
+end
+macro irrational(sym, def)
+    irrational(sym, Float64(eval(def)), def)
```
Comment (simeonschaub) [inline base/irrationals.jl:180]: We definitely don't want to call `eval` inside a macro. Why not:
```suggestion
    irrational(sym, :(Float64(def)), def)
```

**Comment by LilithHafner** [comment]
I don't think the @irrational macro is currently tested at all (`grep -r @irrational test`), and I'm not sure how to write a test for it. I'd appreciate advice.

**Comment by KristofferC** [comment]
Should the existing `@irrational` usages switch to this form now?

**Comment by LilithHafner** [comment]
Bump

**Comment by LilithHafner** [comment]
Bump. Is there anything else this needs?

**Pull request merged**